firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLitePersistence.java

```diff
@@ -202,14 +207,31 @@ void runTransaction(String action, Runnable operation) {
     return value;
   }
 
+  public static void clearPersistence(Context context, DatabaseId databaseId, String persistenceKey)
+      throws FirebaseFirestoreException {
+    String databaseName = SQLitePersistence.databaseName(persistenceKey, databaseId);
+    String sqLitePath = context.getDatabasePath(databaseName).getPath();
+    String journalPath = sqLitePath + "-journal";
```

Comment:
I think you'll also need to delete the WAL. (sqlitePath + "-wal"). This only applies to android 9/pie and newer devices, but should be safe enough to do unconditionally.

https://source.android.com/devices/tech/perf/compatibility-wal
https://www.sqlite.org/tempfiles.html#write_ahead_log_wal_files